### PR TITLE
DOP-4474: redirects for offline part 1

### PR DESCRIPTION
**Pull request by schmalliso**
### Stories/Links:

DOP-4474

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

File: infrastructure/ecs-main/buckets.yml [hunk 1]
```diff
@@ -25,83 +25,149 @@ Resources:
...
                 HostName: ${self:custom.site.host.${self:provider.stage}}
                 ReplaceKeyPrefixWith: ${self:custom.site.prefix.${self:provider.stage}}/atlas/cli/v1.2
             - RoutingRuleCondition:
-                KeyPrefixEquals: ${self:custom.site.prefix.${self:provider.stage}}/atlas/cli/v1.2.0
+                KeyPrefixEquals: ${self:custom.site.prefix.${self:provider.stage}}/atlas/cli/v1.1
               RedirectRule:
                 Protocol: "https"
                 HostName: ${self:custom.site.host.${self:provider.stage}}
-                ReplaceKeyPrefixWith: ${self:custom.site.prefix.${self:provider.stage}}/atlas/cli/v1.2
+                ReplaceKeyPrefixWith: ${self:custom.site.prefix.${self:provider.stage}}/atlas/cli/v1.1
```
Comment (mayaraman19) [inline infrastructure/ecs-main/buckets.yml:38]: a little confused on these changes - won't this redirect /v1.2 to itself and same with v1.1?

File: infrastructure/ecs-main/buckets.yml [hunk 2]
```diff
@@ -25,83 +25,149 @@ Resources:
...
+                HostName: ${self.custom.site.host.${self:provider.stage}}
+                ReplaceKeyPrefixWith: ${self:custom.site.prefix.${self:provider.stage}}/kubernetes-operator/stable
+            - RoutingRuleCondition:
+                KeyPrefixEquals: ${self:custom.site.prefix.${self:provider.stage}}/kubernetes-operator/v1.0
+              RedirectRule:
+                Protocol: "https"
+                HostName: ${self.custom.site.host.${self:provider.stage}}
+                ReplaceKeyPrefixWith: ${self:custom.site.prefix.${self:provider.stage}}/kubernetes-operator/stable
+            - RoutingRuleCondition:
+                KeyPrefixEquals: ${self:custom.site.prefix.${self:provider.stage}}/kubernetes-operator/v1.1
```
Comment (mayaraman19) [inline infrastructure/ecs-main/buckets.yml:142]: do we also need to add rules for k8s v1.2 through v1.9? i see those on the spreadsheet but (unless i'm not looking correctly) i don't see those here!

File: infrastructure/ecs-main/buckets.yml [hunk 3]
```diff
@@ -25,83 +25,149 @@ Resources:
...
+              RedirectRule:
+                Protocol: "https"
+                HostName: ${self.custom.site.host.${self:provider.stage}}
+                ReplaceKeyPrefixWith: ${self:custom.site.prefix.${self:provider.stage}}/kubernetes-operator/stable
+            - RoutingRuleCondition:
+                KeyPrefixEquals: ${self:custom.site.prefix.${self:provider.stage}}/mongocli/v0
```
Comment (mayaraman19) [inline infrastructure/ecs-main/buckets.yml:148]: (tiny micro nit) perhaps make this `v0.` for consistency with the `v0.` above?

File: infrastructure/ecs-main/buckets.yml [hunk 4]
```diff
@@ -25,83 +25,149 @@ Resources:
...
+              RedirectRule:
+                Protocol: "https"
+                HostName: ${self.custom.site.host.${self:provider.stage}}
+                ReplaceKeyPrefixWith: ${self:custom.site.prefix.${self:provider.stage}}/mongocli/stable
```
Comment (mayaraman19) [inline infrastructure/ecs-main/buckets.yml:152]: again, unless I'm not looking correctly, I don't see redirects for mongo cli v1.0.0-1.24

File: infrastructure/ecs-main/buckets.yml (hunk 1 above)
Comment (schmalliso) [reply infrastructure/ecs-main/buckets.yml:38]: Yes, but it'll also capture v1.2.x and v1.21 and replace them with v1.2, which is the outcome we want (same with v1.1*)

File: infrastructure/ecs-main/buckets.yml (hunk 4 above)
Comment (schmalliso) [reply infrastructure/ecs-main/buckets.yml:152]: mongocli v1.0.0-v1.24 build with Snooty, so we don't have offline versions of them currently! Marvellous attention to detail though :)

File: infrastructure/ecs-main/buckets.yml (hunk 2 above)
Comment (schmalliso) [reply infrastructure/ecs-main/buckets.yml:142]: Ooh good catch, it should just be v1 -- fixing.

File: infrastructure/ecs-main/buckets.yml [hunk 5]
```diff
@@ -25,83 +25,149 @@ Resources:
...
+                HostName: ${self.custom.site.host.${self:provider.stage}}
+                ReplaceKeyPrefixWith: ${self:custom.site.prefix.${self:provider.stage}}/bi-connector/current
+            - RoutingRuleCondition:
+                KeyPrefixEquals: ${self:custom.site.prefix.${self:provider.stage}}/kubernetes-operator/v0.
+              RedirectRule:
+                Protocol: "https"
+                HostName: ${self.custom.site.host.${self:provider.stage}}
+                ReplaceKeyPrefixWith: ${self:custom.site.prefix.${self:provider.stage}}/kubernetes-operator/stable
+            - RoutingRuleCondition:
+                KeyPrefixEquals: ${self:custom.site.prefix.${self:provider.stage}}/kubernetes-operator/v1.0
```
Comment (mayaraman19) [inline infrastructure/ecs-main/buckets.yml:136]: can probaby remove this rule, since we have the `/v1` prefix now

**Review by mayaraman19 — APPROVED** [review]
looks good, left a small comment about removing redundancy but that's non-blocking

**Pull request merged**